src/convert-lara-answers/convert-interactive-state.test.ts

```diff
@@ -0,0 +1,147 @@
+};
+
+const laraAnswerAnonymous: ILARAAnonymousAnswerReportHash & ILARAInteractiveRunStateReportHash = {
+  version: '1',
+  question_id: 'mw_interactive_210348',
+  resource_url: 'https://authoring.staging.concord.org/activities/20624',
+  class_info_url: null,
+  report_state: '{"version":1,"mode":"report","authoredState":null,"interactiveState":"{\\"version\\":1,\\"exercises\\":{\\"summary\\":{\\"current\\":\\"# Get summary statistics for the \\\\\\"reusable_trays\\\\\\" data set \\\\nsummary(reusable_trays)\\\\n\\",\\"submitted\\":\\"\\"}},\\"lara_options\\":{\\"reporting_url\\":\\"https://concord.shinyapps.io/reus_summary/\\"}}"}',
+  platform_id: null,
+  source_key: 'authoring.staging.concord.org',
+  tool_id: 'https://authoring.staging.concord.org',
+  answer_text: null,
+  run_key: '6613df87-efa7-4286-9c88-7ba738c2d9e7',
+  platform_user_id: '6613df87-efa7-4286-9c88-7ba738c2d9e7',
+  remote_endpoint: null,
+  resource_link_id: null,
+  context_id: null,
+  answer: '{"version":1,"mode":"report","authoredState":null,"interactiveState":"{\\"version\\":1,\\"exercises\\":{\\"summary\\":{\\"current\\":\\"# Get summary statistics for the \\\\\\"reusable_trays\\\\\\" data set \\\\nsummary(reusable_trays)\\\\n\\",\\"submitted\\":\\"\\"}},\\"lara_options\\":{\\"reporting_url\\":\\"https://concord.shinyapps.io/reus_summary/\\"}}"}',
+  created: '2021-10-13 16:22:45 UTC',
+  id: 'interactive_run_state_62278',
+  question_type: 'iframe_interactive',
+  submitted: null,
+  tool_user_id: '2372',
+  type: 'interactive_state'
+};
+
+describe("convert interactive state", () => {
+  it("copies all the expected fields of the logged in variant of the answer", () => {
+    const result = convertInteractiveState({
+      oldAnswer: laraAnswerLoggedIn,
+      newQuestion,
+      oldSourceKey: "authoring.concord.org",
+      newSourceKey: "activity-player.concord.org",
+      additionalMetadata: {}
+    });
+
+    expect(result).toEqual({
+      report_state: '{"version":1,"mode":"report","authoredState":null,"interactiveState":"{\\"data\\":[[\\"a\\",\\"Table\\",\\"that\\",\\"saves\\"],[\\"b\\",null,\\"state\\",null],[\\"c\\",null,null,null]]}"}',
+      question_type: 'iframe_interactive',
+      submitted: null,
+      tool_user_id: '1406',
+      id: 'converted-authoring.concord.org-answers-interactive_run_state_78721',
+      answer: '{"version":1,"mode":"report","authoredState":null,"interactiveState":"{\\"data\\":[[\\"a\\",\\"Table\\",\\"that\\",\\"saves\\"],[\\"b\\",null,\\"state\\",null],[\\"c\\",null,null,null]]}"}',
+      type: 'interactive_state',
+      resource_link_id: '2462',
+      remote_endpoint: 'https://learn.staging.concord.org/dataservice/external_activity_data/933e2f3f-13b2-4119-b3b9-97485aaa8c58',
+      tool_id: 'activity-player.concord.org',
+      class_info_url: 'https://learn.staging.concord.org/api/v1/classes/32',
+      run_key: '19bfcb93-ccbf-4f53-84a7-296294821954',
+      question_id: 'managed_interactive_448816',
```

Comment:
The question_id is changed here from 'mw_interactive_212649'.
I would have expected it to stay the same since we are not converting existing interactives. 
But I guess it is just because of the mocking of the `new_question`.

Reply:
Yes, in the real-world scenario, the passed question would be an old question in fact. So, the ID will be updated to the same value.